src/Opdex.Platform.Infrastructure/Data/DbContext.cs

```diff
@@ -2,64 +2,78 @@
     {
-        var command = BuildCommandDefinition(query);
         await using var connection = _databaseSettings.Create();
-        return await connection.QueryAsync<TEntity>(command);
+        return await Execute(query, async c => await connection.QueryAsync<TEntity>(c));
     }
 
     public async Task<IEnumerable<TReturn>> ExecuteQueryAsync<TFirst, TSecond, TReturn>(DatabaseQuery query,
                                                                                         Func<TFirst, TSecond, TReturn> map,
                                                                                         string splitOn)
     {
-        var command = BuildCommandDefinition(query);
         await using var connection = _databaseSettings.Create();
-        return await connection.QueryAsync(command, map, splitOn);
+        return await Execute(query, async c => await connection.QueryAsync(c, map, splitOn));
     }
 
     public async Task<TEntity> ExecuteFindAsync<TEntity>(DatabaseQuery query)
     {
-        var command = BuildCommandDefinition(query);
         await using var connection = _databaseSettings.Create();
-        return await connection.QuerySingleOrDefaultAsync<TEntity>(command);
+        return await Execute(query, async c => await connection.QuerySingleOrDefaultAsync<TEntity>(c));
     }
 
     public async Task<TEntity> ExecuteScalarAsync<TEntity>(DatabaseQuery query)
     {
-        var command = BuildCommandDefinition(query);
         await using var connection = _databaseSettings.Create();
-        return await connection.ExecuteScalarAsync<TEntity>(command);
+        return await Execute(query, async c => await connection.ExecuteScalarAsync<TEntity>(c));
     }
 
     public async Task<int> ExecuteCommandAsync(DatabaseQuery query)
     {
-        var command = BuildCommandDefinition(query);
         await using var connection = _databaseSettings.Create();
-        return await connection.ExecuteAsync(command);
+        return await Execute(query, async c => await connection.ExecuteAsync(c));
     }
 
-    private static CommandDefinition BuildCommandDefinition(DatabaseQuery query)
+    private async Task<TR> Execute<TR>(DatabaseQuery query, Func<CommandDefinition, Task<TR>> action)
     {
-        return new CommandDefinition(query.Sql, query.Parameters, commandType: query.Type, cancellationToken: query.Token);
+        var command = new CommandDefinition(query.Sql, query.Parameters, commandType: query.Type, cancellationToken: query.Token);
+
+        var stopwatch = new Stopwatch();
+        stopwatch.Start();
+        var result = await action.Invoke(command);
+        stopwatch.Stop();
+
+        using (_logger.BeginScope(new Dictionary<string, object>
+               {
+                   { "Command", command.CommandText }
+               }))
+        {
+            _logger.LogDebug("Executed database query in {ExecutionTimeMs} ms", stopwatch.ElapsedMilliseconds);
```

Comment:
Idk but I'll ask anyways, you sure that `{ExecutionTimeMs}` will be replaced and `{0}` shouldn't be used instead?